airbyte-integrations/connectors/source-salesforce/source_salesforce/spec.json

```diff
@@ -43,6 +43,14 @@
         "type": "string",
         "enum": ["BULK", "REST"],
         "default": "BULK"
+      },
+      "wait_timeout": {
```

Comment:
IMO this option should not be exposed - it is an implementation detail. See the [UX handbook](https://app.gitbook.com/o/-MDx7_cfclXFzHgniOWQ/s/-MHZYvlJGSuz4DxQs3lX/c/W4juESqgNddJ5jvXRACm/connector-development/ux-handbook#input-configuration) for more context behind why it's not a good idea to expose it. Maybe instead we can use a dynamically increasing wait time for each job.

Reply:
Thus I forwarded this PR to Airbyte team)
Your idea with autoscaling of waiting timeout is pretty but we can reproduce this case with long responses locally. For more details I added more informative log messages for troubleshooting of possible similar cases.